### ISSUE TITLE
Lint with flake8

### ISSUE THREAD

**Issue by ColdHeat** [CTFd/CTFd#823]
We should lint with flake8 so that we can catch some bugs at build time. 

A safe start would be:
```
flake8 --ignore E501 CTFd/
```

**Comment by jsfan** [comment]
I am planning to look into this in the next days. However, you suggest that you would like this configuration to be able to catch bugs at build time. So, is this what you are after?

1. Run pylint/flake8/pycodestyle over the code and address all errors
2. Integrate a call to the same tool in the Travis configuration to make sure violations are caught at buid time

Probably rather independent of the approach, a PR would likely have to be merged pass because it would certainly develop conflicts quickly. How do we coordinate that?

**Comment by ColdHeat** [comment]
Sorry I think build time is the wrong word to use. I think at test time is a better phrase. 

I think I can start a 2.1 branch and then you can PR in the flake8 changes and slowly remove the violations. I imagine most of them are going to be about unused imports.

**Comment by jsfan** [comment]
There definitely are a good number of violations but I have not made any effort yet to determine how many different ones it might be.

I'm happy to do the manual cleanup with a good number of small pull requests which should allow quick merging without too much hassle.

I suspect the end goal still is to enforce these same rules in Travis eventually, right?

**Comment by ColdHeat** [comment]
Yes the build should fail on flake8 errors just like how it fails on pycodestyle errors now.

**Comment by jsfan** [comment]
I've managed to get through all the unused imports in the CTFd package and fully rebased against master. As you have not created a 2.1 branch, yet, I'm unsure if I should send a PR.

What is still outstanding is all other flake8 errors (the Travis config will only check for unused imports for now) and going through the unit tests as well.

**Comment by ColdHeat** [comment]
@jsfan here you go: https://github.com/CTFd/CTFd/tree/2.1.0-dev 😄

**Comment by jsfan** [comment]
That was quick! The PR will need to be squashed. I just did the cleanup in stages to see if anything was breaking.

I've also had a look at a few other violations and noticed that my workarounds for some only seemigly unused imports have resulted in unused variables. AFAIK, flake8 does not allow declaring a local exception with a code comment the way pylint does. So, I'll have to think about how to address these remaining issues.

**Comment by ColdHeat** [comment]
I haven’t tried it myself at the moment but I believe you can use # noqa
and the exceptions you want to ignore?

**Comment by jsfan** [comment]
Ah, cool. I expected that there would be a way (knowing it's possible in pylint) but when I looked at the flake8 docs yesterday, I couldn't find anything. I might have just looked in the wrong place then.

**Comment by ColdHeat** [comment]
Should be resolved with #842
